### PR TITLE
General code healt, nodiscard, range loops etc

### PR DESCRIPTION
**Pull request by baldersheim**
@vekterli PR
No semantic changes, just pure code health

### CODE REVIEW

File: storage/src/vespa/storage/bucketdb/bucketmanager.cpp [hunk 1]
```diff
@@ -419,23 +402,17 @@ bool BucketManager::onRequestBucketInfo(
...
-        for (uint32_t i = 0; i < cmd->getBuckets().size(); i++) {
-            BucketMap entries(_component.getBucketDatabase(bucketSpace).getAll(
-                                    cmd->getBuckets()[i],
-                                    "BucketManager::onRequestBucketInfo"));
-            for (BucketMap::iterator it = entries.begin();
-                 it != entries.end(); ++it)
-            {
-                info.push_back(api::RequestBucketInfoReply::Entry(
-                            it->first, it->second->getBucketInfo()));
+        for (auto i : cmd->getBuckets()) {
```
Comment (vekterli) [inline storage/src/vespa/storage/bucketdb/bucketmanager.cpp:408]: Consider using a non-generic loop variable name

File: storage/src/vespa/storage/bucketdb/bucketmanager.cpp [hunk 2]
```diff
@@ -419,23 +402,17 @@ bool BucketManager::onRequestBucketInfo(
...
+            BucketMap entries(_component.getBucketDatabase(bucketSpace).getAll(i, "BucketManager::onRequestBucketInfo"));
+            for (auto & entrie : entries) {
```
Comment (vekterli) [inline storage/src/vespa/storage/bucketdb/bucketmanager.cpp:410]: `s/entrie/entry`

**Review by vekterli — APPROVED** [review]

File: storage/src/vespa/storage/bucketdb/bucketmanager.cpp (hunk 1 above)
Comment (baldersheim) [reply storage/src/vespa/storage/bucketdb/bucketmanager.cpp]: Fixed

File: storage/src/vespa/storage/bucketdb/bucketmanager.cpp (hunk 2 above)
Comment (baldersheim) [reply storage/src/vespa/storage/bucketdb/bucketmanager.cpp]: Fixed

**Pull request merged**